allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -1135,7 +1141,34 @@ def stimulus_templates(self) -> pd.DataFrame:
                     image array of warped stimulus image
 
         """
-        return self._stimuli.templates.value.to_dataframe()
+        return self._stimuli.templates.value[0].to_dataframe()
+
+    @property
+    def stimulus_template_movie(self) -> pd.DataFrame:
+        """Get stimulus templates movie for the behavior session.
+
+        Returns None if no stimulus movie is available.
+
+        Returns
+        -------
+        pd.DataFrame or None
+            A pandas DataFrame object containing the individual frames for the
+            movie shown during this experiment.
+
+            dataframe columns:
+                frame_number [index]: (int)
+                    Frame number in movie
+                unwarped: (array of int)
+                    image array of unwarped stimulus movie frame
+                warped: (array of int)
+                    image array of warped stimulus movie frame
+
+        """
+        if len(self._stimuli.templates.value) > 1:
```

Comment:
Instead of requiring that the movie must be the 2nd item in the list, can we add a property to the `Stimuli/Templates` class for the movie? Then the check will become

```
fingerprint_stimulus = self._stimuli.fingerprint_stimulus
if fingerprint_stimulus is not None:
    fingerprint_stimulus = fingerprint_stimulus.to_dataframe(...)
return fingerprint_stimulus
```

Reply:
So instead of the `Stimuli` object holding one Templates set, it would hold two with the option being the fingerprint stimulus?

My thinking around what you're asking would be to make a small inherited class for `Templates` that overwrites `from_stimulus_file` and attempts to load the movie. If there is no movie, the method returns None and a `Templates` instance otherwise. The `Stimuli` method `from_stimulus_file` would then always attempt to call this subclass along with current templates and store it in, as you suggest, the fingerprint_stimulus/templates variable. Does that make sense/sound good?

Reply:
Done. Changed to a dictionary for the input as we discussed. Added code to parse the keys to identify which is a "movie" or other templates.